src/cards/wakatime-card.js

```diff
@@ -107,6 +143,71 @@ const renderWakatimeCard = (stats = {}, options = { hide: [] }) => {
     iconColor,
   });
 
+  let finalLayout = "";
+
+  let width = 440;
+
+
+  // RENDER COMPACT LAYOUT
+  if (layout === "compact") {
+    width = width + 50;
+    height = 90 + Math.round(languages.length / 2) * 25;
+
+    // progressOffset holds the previous language's width and used to offset the next language
+    // so that we can stack them one after another, like this: [--][----][---]
+    let progressOffset = 0;
+    const compactProgressBar = languages
+      .map((lang) => {
+
+        // const progress = (width * lang.percent) / 100;
+        const progress = ((width - 50) * lang.percent) / 100;
+
+        const languageColor = languageColors[lang.name] || '#858585';
+
+        const output = `
+          <rect
+            mask="url(#rect-mask)" 
+            data-testid="lang-progress"
+            x="${progressOffset}" 
+            y="0"
+            width="${progress}" 
+            height="8"
+            fill="${languageColor}"
+          />
+        `;
+        progressOffset += progress;
+        return output;
+      })
+      .join("");
+    
+    finalLayout = `
+      <mask id="rect-mask">
+      <rect x="25" y="0" width="${width - 50
+      }" height="8" fill="white" rx="5" />
+      </mask>
+      ${compactProgressBar}
+      ${createLanguageTextNode({
+        x: 0,
+        y: 25,
+        langs: languages,
+        totalSize: 100,
+      }).join("")}
+    `;
+  } else {
+    finalLayout = FlexLayout({
+      items: statItems.length
+        ? statItems
+        : [
+          noCodingActivityNode({
+            color: textColor,
+            text: i18n.t("wakatimecard.nocodingactivity"),
+          }),
+        ],
+      gap: lheight,
+      direction: "column",
+    }).join("");
+  }
+
```

Comment:
Hmmmmmm 🤔 really need to think how can we refactor this code into our reusable Card.js

Reply:
How can we do this?